transports/janus_http.c

```diff
@@ -1536,7 +1536,9 @@ static MHD_Result janus_http_handler(void *cls, struct MHD_Connection *connectio
 		const char *secret = MHD_lookup_connection_value(connection, MHD_GET_ARGUMENT_KIND, "apisecret");
 		const char *token = MHD_lookup_connection_value(connection, MHD_GET_ARGUMENT_KIND, "token");
 		gboolean secret_authorized = FALSE, token_authorized = FALSE;
-		if(!gateway->is_api_secret_needed(&janus_http_transport) && !gateway->is_auth_token_needed(&janus_http_transport)) {
+		gboolean is_api_secret_needed = gateway->is_api_secret_needed(&janus_http_transport);
+		gboolean is_auth_token_needed = gateway->is_auth_token_needed(&janus_http_transport);
+		if(!is_api_secret_needed && !is_auth_token_needed) {
```

Comment:
Same here: if defaults for `secret_authorized` and `token_authorized` are set to `TRUE` if the related mechanism is disabled, then the checks are simpler. In this case, I agree the check below must get back to an `&&` rather than an `||` as in the core.

Reply:
So the code based on the suggestion might look like

```c
		gboolean secret_authorized = !gateway->is_api_secret_needed(&janus_http_transport), token_authorized = gateway->is_auth_token_needed(&janus_http_transport);
		if(secret_authorized && token_authorized) {
			/* Nothing to check */
		} else {
			if(!secret_authorized && gateway->is_api_secret_valid(&janus_http_transport, secret)) {
				/* API secret is valid or disabled */
				secret_authorized = TRUE;
			}
			if(!token_authorized && gateway->is_auth_token_valid(&janus_http_transport, token)) {
				/* Token is valid or disabled */
				token_authorized = TRUE;
			}
			/* We consider a request authorized if either the proper API secret or a valid token has been provided */
			if(!secret_authorized && !token_authorized) {
				response = MHD_create_response_from_buffer(0, NULL, MHD_RESPMEM_PERSISTENT);
				janus_http_add_cors_headers(msg, response);
				ret = MHD_queue_response(connection, MHD_HTTP_FORBIDDEN, response);
				MHD_destroy_response(response);
				goto done;
			}
		}
```